### PR TITLE
Task #5228 - Exort csv for user and book by query

### PR DESCRIPTION
**Pull request by taylq**
https://edu-redmine.framgia.vn/issues/5228?issue_count=25&issue_position=22&next_issue_id=5227&prev_issue_id=5229

https://edu-redmine.framgia.vn/issues/5229?issue_count=25&issue_position=21&next_issue_id=5228&prev_issue_id=5230

https://edu-redmine.framgia.vn/issues/5240?issue_count=25&issue_position=10&next_issue_id=5239&prev_issue_id=5241

### CODE REVIEW

**Comment by taylq** [comment]
ready

**Comment by taylq** [comment]
ready

**Comment by taylq** [comment]
ready

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -3,7 +3,12 @@ class UsersController < BaseController
     before_action :find_user, except: %i(index new create)
 
     def index
-      @users = User.select_attr.page(params[:page]).per(Settings.per_page)
+      @users = User.select_attr.page(params[:page]).per(Settings.per_page).search params[:search]
+      respond_to do |format|
+        format.html {}
+        format.csv { send_data User.search(params[:search]).to_csv }
+        format.xls { send_data User.search(params[:search]).to_csv }
```
[the remark this answers is not in the document]
Comment (taylq) [reply app/controllers/admin/users_controller.rb]: 2 cái này e dùng chung method to_csv ở trong model ạ.

**Comment by taylq** [comment]
ready

File: app/models/book.rb [hunk 2]
```diff
@@ -26,5 +26,14 @@ def search search
         Book.select_attr
       end
     end
+
+    def to_csv
+      CSV.generate do |csv|
+        csv << column_names
```
Comment (huongnguyenmta) [inline app/models/book.rb]: Đặc tả một số trường cần thiết để export thôi em

File: app/models/user.rb [hunk 3]
```diff
@@ -50,6 +50,15 @@ def search search
         User.select_attr
       end
     end
+
+    def to_csv
+      CSV.generate do |csv|
+        csv << column_names
```
Comment (huongnguyenmta) [inline app/models/user.rb]: tương tự

**Pull request merged**